schema-language/src/main/java/org/coursera/courier/grammar/CourierSchemaParser.java

```diff
@@ -469,9 +473,24 @@ private UnionDataSchema parseUnion(
     return schema;
   }
 
-  private RecordDataSchema parseRecord(
+  private RecordDataSchema parseKey(
```

Comment:
We likely have to copy `RecordDataSchema` since it's final or decorate it with additional annotations so we don't erase type information of the new `key` keyword. It's also one possible place we can add some validation to make sure the courier document conforms to spec